### ISSUE TITLE
[Bug] Azure changes redirect_uri from http to https causing TcpInterceptor to fail in parsing the response

### ISSUE THREAD

**Issue by jgoyvaerts** [AzureAD/microsoft-authentication-library-for-dotnet#1266]
**Which Version of MSAL are you using ?** 
4.1

**Platform**
netcore

**What authentication flow has the issue?**
* Desktop / Mobile
    * [x] Interactive
    * [ ] Integrated Windows Auth
    * [ ] Username Password
    * [ ] Device code flow (browserless)
* Web App
    * [ ] Authorization code
    * [ ] OBO
* Web API
    * [ ] OBO

Other? - please describe;

**Is this a new or existing app?**
New app

**Repro**

```csharp
var clientId = "<CLIENT-ID>";
var authority = "https://login.windows.net/<TENANT-ID>";
var scopes = new[] { "https://vault.azure.net/.default" };
IPublicClientApplication app = PublicClientApplicationBuilder.Create(clientId)
                .WithAuthority(new Uri(authority), true)
                .WithRedirectUri("http://localhost:4001")
                .Build();

var result = app.AcquireTokenInteractive(scopes).ExecuteAsync().GetAwaiter().GetResult();
```

**Expected behavior**
A token should be received

**Actual behavior**
An exception is triggered 
**Microsoft.Identity.Client.MsalClientException: 'Could not extract the query from the authorization response - check Pii enabled logs for details'**

**Possible Solution**
After debugging this a bit, it turns out that while MSAL is sending the correct request to Azure (redirect_uri=http://localhost:4001), Azure then redirects to httpS://localhost:4001, which causes the TcpInterceptor to fail because it receives an ssl handshake request instead of a http request.

This might be a bug on the Azure side (why is it changing http to https?), but since this is the library which is exposing the bug I'm reporting it here.

**Comment by jmprieur** [comment]
Which redirect URI did register in the app registration portal, @jgoyvaerts ?

**Comment by jgoyvaerts** [comment]
I registered both http://localhost and http://localhost:4001

**Comment by jmprieur** [comment]
Do you need to specify a port? Did you try with just "http://localhost" ?

**Comment by jgoyvaerts** [comment]
Same behavior with just http://localhost as redirect uri in the app registration portal

**Comment by bgavrilMS** [comment]
I can't repro this. Here's what I tried: 

1. Create a new app registration 
2. Set "http://localhost:4001" as a redirect uri
3. Add API permission for KeyVault
4. Grant admin access for API permissions
5. Run the code you provided 

Actual: authentication was successful.

![image](https://user-images.githubusercontent.com/12273384/61290738-80551900-a7c4-11e9-8edf-75ecefda5e02.png)

We also run this scenario is our lab every night. 

Can you think of anything else that might be affecting your scenario?

**Comment by jgoyvaerts** [comment]
Followed your exact same steps, even with a new app I get the same behavior.

Below you can find the app registration manifest:
```
{
	"id": "658c3f5b-bd4b-44e7-a152-9d3599cdf26b",
	"acceptMappedClaims": null,
	"accessTokenAcceptedVersion": null,
	"addIns": [],
	"allowPublicClient": null,
	"appId": "9a6dd9c8-4dec-4ee5-b4e0-b668d623c03b",
	"appRoles": [],
	"oauth2AllowUrlPathMatching": false,
	"createdDateTime": "2019-07-16T12:07:39Z",
	"groupMembershipClaims": null,
	"identifierUris": [],
	"informationalUrls": {
		"termsOfService": null,
		"support": null,
		"privacy": null,
		"marketing": null
	},
	"keyCredentials": [],
	"knownClientApplications": [],
	"logoUrl": null,
	"logoutUrl": null,
	"name": "TestApp",
	"oauth2AllowIdTokenImplicitFlow": false,
	"oauth2AllowImplicitFlow": false,
	"oauth2Permissions": [],
	"oauth2RequirePostResponse": false,
	"optionalClaims": null,
	"orgRestrictions": [],
	"parentalControlSettings": {
		"countriesBlockedForMinors": [],
		"legalAgeGroupRule": "Allow"
	},
	"passwordCredentials": [
		{
			"customKeyIdentifier": null,
			"endDate": "2299-12-30T23:00:00Z",
			"keyId": "6d578450-24a4-4379-a46d-3394474cfa5f",
			"startDate": "2019-07-16T12:08:01.145Z",
			"value": null,
			"createdOn": "2019-07-16T12:08:01.3347462Z",
			"hint": "w_n",
			"displayName": "Test"
		}
	],
	"preAuthorizedApplications": [],
	"publisherDomain": "virteo.be",
	"replyUrlsWithType": [
		{
			"url": "http://localhost:4001",
			"type": "InstalledClient"
		}
	],
	"requiredResourceAccess": [
		{
			"resourceAppId": "cfa8b339-82a2-471a-a3c9-0fc0be7a4093",
			"resourceAccess": [
				{
					"id": "f53da476-18e3-4152-8e01-aec403e6edc0",
					"type": "Scope"
				}
			]
		},
		{
			"resourceAppId": "00000003-0000-0000-c000-000000000000",
			"resourceAccess": [
				{
					"id": "e1fe6dd8-ba31-4d61-89e7-88639da4683d",
					"type": "Scope"
				}
			]
		}
	],
	"samlMetadataUrl": null,
	"signInUrl": null,
	"signInAudience": "AzureADMyOrg",
	"tags": [],
	"tokenEncryptionKeyId": null
}
```

**Comment by bgavrilMS** [comment]
Are you using a .net core app? What .net core version?

**Comment by jgoyvaerts** [comment]
.Net core 2.2
Not sure how that's relevant since it's azure who's changing the http to https?

The url which is being requested on azure side:

https://login.microsoftonline.com/{TENANTID}/oauth2/v2.0/authorize?scope=https%3A%2F%2Fvault.azure.net%2F.default+offline_access+openid+profile&response_type=code&client_id=9a6dd9c8-4dec-4ee5-b4e0-b668d623c03b&redirect_uri=http%3A%2F%2Flocalhost%3A4001&client-request-id=c87d3633-4a93-484a-9ae1-04aa61c94812&x-client-SKU=MSAL.NetCore&x-client-Ver=4.0.0.0&x-client-OS=Microsoft+Windows+10.0.17134+&prompt=select_account&code_challenge=MnOO8DJOGmh7yQPSYPUX_OMQg8mPHlS4ZS3KBxxKW6I&code_challenge_method=S256&state=bcf98606-9093-43a2-a0d5-c0bd63d4586f5f0ef9c9-2767-43c0-85cd-e8110fd5169f

The url azure is redirecting to:

https://localhost:4001/?code=OAQABAAIAAAAP0wLlqdLVToOpA4kwzSnxjp9Cp6bMzC9KCRFULQoqxQMQmrFS3K8tPJOBLPuPul5dyWOF1C5TKDbmPIGhMWEd9p79ohuOSWcV4PyWE8pThaViQqVdP6laBX9IS8-zUIxDYc4uJJRqeB_muUnXhSZRXakl4bziku3dNHKcjqclLkT6ApAerV8fop8qJp0-lYfxxzJxLa3kSyZyM38FP4N4nOBUlZsJVGWTRyZ8bF1QyoT0kbhLhQ0_DeJuOmmc3QbtJF9xv5YzaVA3Q96uEOL1gFzv-pTXbK4vUWz1-M1F9DRZwQG_XcKQRP4P9Lbv4A7iH2gU0F6lu8Zf4vWysj7q1Fq9Y8QQMmlkJq3xGlzx2KqqxHTHmDEnazO_QhgZaEq4r6j_GkPi1tYCrbHrfLcu9Hq-RjiGM5PJfQh6r2k-hjZohDjbZsSmKtkCeFiQARFoZKwvmevr2dgB41RGgFeewAML88iql16bJQ6f3QyNKKrnNa268GL45iT-XA61SjbcdgqEHDygFo2lVPk1UjFdB9wf2Cp0OHK2mfG5PmMkWthhvj6Gk9YqEVj7HGuKnyyOzmXvD6pWZ-vl_YRl8RNrdxR5jvwo0cElyhmSLk9MAlOkCnmRRmf_idRWy_oMVPbWkl88JQME6oXikvMZSUYgHCz4R7Y9ShhR-f_ny5b9tBJfU46Dvj_tvoNYbyawmmmGPodtRJGr8QuAXLIiRCKMytKleEfSn93LvbE6-o0rl4aiSkN7xlPxbNh4eU0kfYMevebrtF3roPb1C_HgrXU-IAA&state=bcf98606-9093-43a2-a0d5-c0bd63d4586f5f0ef9c9-2767-43c0-85cd-e8110fd5169f&session_state=07d352a9-a16b-4ac4-8ace-4ee2053bf195

Note the difference between http and https.

**Comment by jmprieur** [comment]
@jgoyvaerts 
where is your application deployed? are you using a proxy?

**Comment by jgoyvaerts** [comment]
I'm running the application locally, from visual studio debugging. No proxies are involved.

I just checked my chrome network tab, the request is first being sent to http but chrome is upgrading the request to https because a different site hosted on my localhost has HSTS enabled (and HSTS is domain based, not domain + port based).

Closing this ticket since it's a local pc issue, sorry for the confusion.
